### PR TITLE
Spawn Sign with your own text in Sandbox

### PR DESCRIPTION
**Pull request by mugg91**
## Status

- **IN DEVELOPMENT**: this PR is still in development, but you would like your changes reviewed.

Will set to "READY" after addressing these:
- Make the game look for bad words in the Sign's text, don't spawn it or censor it.
- Support line breaks with /n like in "TutorialCTF01.as".
- Make sign hittable by builder.
- Default to the shaggy chocolate text when no text is found after "!sign".
- Fix that text can be out of bounds if a word is too long.
- Fix that an icon can be out of bounds. 

## Description

Before this change, in Sandbox, you could type "!sign" and it spawned a sign saying "The big brown fox jumped over the shaggy chocolate". That's all text you could have. "!sign blah" didn't do anything.

This change makes it so !sign lets you put your own text when spawning a sign in Sandbox.
Type in "!sign your text blah blah $icon$".

https://i.imgur.com/0TFCMoa.png

Let me know what you think.

### CODE REVIEW

**Comment by mugg91** [comment]
For future reference, the current build lets you spawn crate with a custom description that is not checked for swear words.

So if I or anyone ever do manage to check for swear words in a self-created custom Sign, then Crates might as well get checked, too.

File: Rules/CommonScripts/ChatCommands.as [hunk 1]
```diff
@@ -302,11 +303,16 @@ bool onServerProcessChat(CRules@ this, const string& in text_in, string& out tex
...
-				else if(tokens[0] == "!coins")
+				else if (tokens[0] == "!coins")
 				{
 					int money = parseInt(tokens[1]);
 					player.server_setCoins(money);
 				}
+				else if (tokens[0] == "!sign")
+				{
+					CMap@ map = getMap();
+					createSign(pos, text_in.substr(tokens[0].length+1, -1));
```
Comment (eps0003) [inline Rules/CommonScripts/ChatCommands.as:314]: The default value for the second argument is -1 so this line can be simplified to the following:
```suggestion
					createSign(pos, text_in.substr(tokens[0].length + 1));
```

**Review by eps0003 — CHANGES_REQUESTED** [review]

**Comment by AsuMagic** [comment]
Will conflict with #1210

**Comment by mugg91** [comment]
This is currently not anywhere near done and there are issues I ran into that I haven't figured out to fix. It would seem some things are engine-side and cannot be done anyway. Please ignore (or close if you must) this PR for the time being.

**Pull request closed without merging**